src/main/scala/com/typesafe/sbt/packager/archetypes/JavaServerApplication.scala

```diff
@@ -242,6 +240,33 @@ object JavaServerAppPackaging extends AutoPlugin {
     }
   }
 
+  private[this] def getEtcTemplateSource(sourceDirectory: File, loader: ServerLoader): java.net.URL = {
+    val suffix = loader match {
+      case Upstart => "-upstart"
+      case SystemV => "-systemv"
+      case Systemd => "-systemd"
+    }
+
+    val overrides = List[File](sourceDirectory / "templates" / (ETC_DEFAULT + suffix), sourceDirectory / "templates" / ETC_DEFAULT)
```

Comment:
Slick implementation. Keeping things backwards compatible ( and easy to use ), but add customized behavior.